src/CppUTestExt/MockSupport.cpp

```diff
@@ -310,20 +310,15 @@ void MockSupport::countCheck()
     UtestShell::getCurrent()->countCheck();
 }
 
-void MockSupport::checkExpectationsOfLastCall()
+void MockSupport::checkExpectations()
 {
+    for(MockNamedValueListNode *p = data_.begin(); p; p = p->next())
+        if(getMockSupport(p) && getMockSupport(p)->lastActualFunctionCall_)
```

Comment:
Do we still need to check the condition `getMockSupport(p)->lastActualFunctionCall_`? if no lastActualFunctionCall_ then we should not go on checking expectation??